### DISCUSSION TITLE
Search history / filter history? feature request?

### DISCUSSION THREAD

**Discussion by JOJ0** [photoprism/photoprism#4537] [Q&A] [answered]
Hi,
before I create a feature request for something I'm missing since I started out with photoprism about 1 to 2 years ago, I'd like to ask the community if either this feature is somewhere and I just didn't find it yet or if a feature request for it or something similar is existing already.

Is it possible to have the search field remember all the filters and searches I used in the past?

Currently I save dozens of searches in an extra notes app because I can't remember them all by heart and certainly it often is tedious to type.

If this feature is not existing, has anybody seen a feature request similar to what I'm proposing already?

Thanks a ton and thanks to devs for maintaining an unbelievably cool tool! ❤️

**Comment by graciousgrey** [comment, marked as answer]
Thank you very much for your feedback!

We plan to implement [smart/conditional albums](https://github.com/photoprism/photoprism/issues/1431) those will be albums based on saved search queries.

We would also like to add a web UI for advanced search that allows users to set additional [filters](https://docs.photoprism.app/user-guide/search/filters/#filter-reference) without having to look them up in the documentation.
